### ISSUE TITLE
clientKey nil exception 1.13.0

### ISSUE THREAD

**Issue by robert-wallis** [ParsePlatform/Parse-SDK-iOS-OSX#896]
This was in the 1.13.0 milestone. #827  But in practice using the framework downloaded from the [releases](https://github.com/ParsePlatform/Parse-SDK-iOS-OSX/releases/tag/1.13.0) causes `'clientKey' should not be nil.` to be thrown.

Specifically with this swift code:
```
let config = ParseClientConfiguration {
    $0.applicationId = kApplicationID
    $0.server = kServer
    $0.clientKey = ""
}
```
Which indicates that the 1.13.0 release does not contain this code #827, or specifically [this commit](https://github.com/ParsePlatform/Parse-SDK-iOS-OSX/commit/177113552e6277aed38a90ed8b4a7ce3f3e06f0f#diff-7407a2df92d06ff9f101eeb944ab5064) because it looks like the property is throwing.

Also in Travis the test `ParseSetupUnitTests::testInitializeWithNilApplicationIdShouldThrowException` is/was `???` ing, maybe it's throwing.  
https://travis-ci.org/ParsePlatform/Parse-SDK-iOS-OSX/jobs/121564192#L1274

What am I doing wrong?

The reason I'm setting it to `nil` or `""` is the iOS code doesn't seem to be creating an installation record on `parse-server`, but the Android code worked after setting the client key to null.

**Comment by nlutsenko** [comment]
Hey @robert-wallis, which platform do you see the problem with?
I've just tested iOS and OS X packages and there it works like a charm.

Can you point me to the specific download that you see the problem with?

Travis would have failed with an error, since xcpretty is also checking for non-zero exit code on the xcodebuild process, which means that all of the tests didn't fail.

**Comment by robert-wallis** [comment]
We are having trouble with an iOS 6 device.  Using the 1.13.0 framework downloaded from https://github.com/ParsePlatform/Parse-SDK-iOS-OSX/releases/tag/1.13.0

To reproduce:

* make an empty Swift project in XCode 7.3.
* Add the Parse framework and Bolts framework from the above link (not CocoaPods). 
* Add the following code in some init location.

```
let config = ParseClientConfiguration {
    $0.applicationId = kApplicationID
    $0.server = kServer
    $0.clientKey = ""
}
```
* Deploy to device.

Crashes with: `'clientKey' should not be nil.` in console.  Thanks for looking into this.  It's possible that the github release code is incorrectly tagged.

**Comment by nlutsenko** [comment]
Ehm, are you sure it's the problem with iOS 6?
The interesting thing is that Swift doesn't support iOS 6 at all.
If you got it to run - that's actually undefined behavior.

Parse SDK also dropped support for iOS 6 a while ago and we are iOS 7.0+ only at this point.

**Comment by robert-wallis** [comment]
Sorry for being an idiot, I meant iOS 9 on an iPhone 6.  Too many things going on right now and I mixed the versions.

**Comment by robert-wallis** [comment]
There must be something wrong with how we set up the framework on iOS because I checked the hash that is tagged 1.13.0 and the assert is no longer there.
https://github.com/ParsePlatform/Parse-SDK-iOS-OSX/blob/dccfe4215ad9d1144089a9c275661a5360466915/Parse/ParseClientConfiguration.m#L67

So it's basically impossible unless we are actually running 1.12.0.
We will try to repro on a different environment.
